### PR TITLE
RFC: boot firmware update support

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Comment by igoropaniuk** [comment]
@mike-sul @ricardosalveti 
there is also one line-change in aklite client (will create it when we reach final decision about this patch):
```
--- a/src/composeappmanager.cc
+++ b/src/composeappmanager.cc
@@ -218,6 +218,7 @@ data::InstallationResult ComposeAppManager::install(const Uptane::Target& target
       LOG_ERROR << "Failed to install OSTree target, skipping Docker Compose Apps";
       return res;
     }
+    const_cast<ComposeAppManager*>(this)->installNotify(target);
   } else {
     LOG_INFO << "Target " << target.sha256Hash() << " is same as current";
     res = data::InstallationResult(data::ResultCode::Numeric::kOk, "OSTree hash already installed, same as current");

```

File: src/libaktualizr/bootloader/bootloader.cc [hunk 1]
```diff
@@ -95,6 +95,65 @@ void Bootloader::updateNotify() const {
   }
 }
 
+void Bootloader::installNotify(const Uptane::Target& target) const {
+  std::string sink;
+  std::string target_hash = target.sha256Hash();
+  std::string version_flag = "bootfirmware_version";
+  // TODO: provide initial /ostree/deploy/lmp/deploy/ from config
+  // instead of hardcoding that here
```
Comment (ricardosalveti) [inline src/libaktualizr/bootloader/bootloader.cc:103]: Yeah, we should have a config entry for the distro/ostree branch and for the version.txt file path.

File: include/libaktualizr/packagemanagerinterface.h [hunk 2]
```diff
@@ -54,6 +54,7 @@ class PackageManagerInterface {
   virtual void completeInstall() const { throw std::runtime_error("Unimplemented"); }
   virtual data::InstallationResult finalizeInstall(const Uptane::Target& target) = 0;
   virtual void updateNotify() {}
+  virtual void installNotify(const Uptane::Target& target) {}
```
Comment (mike-sul) [inline include/libaktualizr/packagemanagerinterface.h:57]: `
/home/mike/work/foundries/projects/aklite-boot/aktualizr/include/libaktualizr/packagemanagerinterface.h: In member function ‘virtual void PackageManagerInterface::installNotify(const Uptane::Target&)’:
/home/mike/work/foundries/projects/aklite-boot/aktualizr/include/libaktualizr/packagemanagerinterface.h:57:52: error: unused parameter ‘target’ [-Werror=unused-parameter]
   virtual void installNotify(const Uptane::Target& target) {}
                                                    ^~~~~~
cc1plus: all warnings being treated as errors

`

**Comment by mike-sul** [comment]
> @mike-sul @ricardosalveti
> there is also one line-change in aklite client (will create it when we reach final decision about this patch):
> 
> ```
> --- a/src/composeappmanager.cc
> +++ b/src/composeappmanager.cc
> @@ -218,6 +218,7 @@ data::InstallationResult ComposeAppManager::install(const Uptane::Target& target
>        LOG_ERROR << "Failed to install OSTree target, skipping Docker Compose Apps";
>        return res;
>      }
> +    const_cast<ComposeAppManager*>(this)->installNotify(target);
>    } else {
>      LOG_INFO << "Target " << target.sha256Hash() << " is same as current";
>      res = data::InstallationResult(data::ResultCode::Numeric::kOk, "OSTree hash already installed, same as current");
> ```

We can isolate/encapsulate the overall change in one place in the scope of aktualizr-lite, so no need for any change to aktualizr project.  For example, the business logic in Bootloader::installNotify() can be represented as a static method of ComposeAppManager or LiteClient (preferrable) and called from `ComposeAppManager::install` or from `LiteClient::install`. The latter is preferable because, as you said, the bootloading logic should be agnostic/independent of package manager type. Also, it sets the path towards getting rid of the dependency on libaktualizr's bootloader at all.

**Pull request closed without merging**